components/analytics-solutions-common/org.wso2.analytics.solutions.common/widgets/DateTimeRangePicker/src/widget/components/CustomTimeRangeSelector.jsx

```diff
@@ -0,0 +1,324 @@
+   * @param{object} date
+   */
+  handleEndTimeChange = (date) => {
+    const { startTime } = this.state;
+    const endTime = date;
+    if (Moment(startTime).isSameOrAfter(Moment(endTime))) {
+      this.setState({
+        invalidDateRange: true,
+        applyButtonBackgroundColor: "#999"
+      });
+    } else {
+      this.setState({
+        invalidDateRange: false,
+        applyButtonBackgroundColor: "#ef6c00",
+        endTime: date
+      });
+    }
+  };
+
+  /*Publishing the custom time range
+  onChangeCustom()=>handleGranularityChangeForCustom(mode, startTime, endTime, granularity)
+  mode:custom
+  granularity:second,minute,hour,day,month,year
+  */
+  publishCustomTimeRange = () => {
+    const { handleClose, onChangeCustom } = this.props;
+    const { customRangeGranularityValue, startTime, endTime } = this.state;
+    handleClose();
+    onChangeCustom("custom", startTime, endTime, customRangeGranularityValue);
+  };
+
+  /**
+   * Change the granularity value of the custom ranges. View of the custom ranges will
+   * change according to the granularity value
+   * @param{string} customGranularityValue:'second','minute','hour' etc
+   */
+  changeCustomRangeGranularity = (customGranularityValue) => {
+    this.props.disableSelectedQuickRangeValue()
+    const { startTime, endTime } = this.state;
+    let invalidDateRange = false;
+    let applyBtnColor = "#ef6c00";
+    switch (customGranularityValue) {
+      case "second":
+        if (Moment(startTime).isSameOrAfter(Moment(endTime), "second")) {
+          invalidDateRange = true;
+          applyBtnColor = "#999";
+        }
+        break;
+      case "minute":
+        if (Moment(startTime).isSameOrAfter(Moment(endTime), "minute")) {
+          invalidDateRange = true;
+          applyBtnColor = "#999";
+        }
+        break;
+      case "hour":
+        if (Moment(startTime).isSameOrAfter(Moment(endTime), "hour")) {
+          invalidDateRange = true;
+          applyBtnColor = "#999";
+        }
+        break;
+      case "day":
+        if (Moment(startTime).isSameOrAfter(Moment(endTime), "day")) {
+          invalidDateRange = true;
+          applyBtnColor = "#999";
+        }
+        break;
+      case "month":
+        if (Moment(startTime).isSameOrAfter(Moment(endTime), "month")) {
+          invalidDateRange = true;
+          applyBtnColor = "#999";
+        }
+        break;
+      case "year":
+        if (Moment(startTime).isSameOrAfter(Moment(endTime), "year")) {
+          invalidDateRange = true;
+          applyBtnColor = "#999";
+        }
+        break;
+      default:
+        break;
+    }
+    this.setState({
+      applyButtonBackgroundColor: applyBtnColor,
+      invalidDateRange: invalidDateRange,
+      customRangeGranularityValue: customGranularityValue
+    });
+  };
+
+  /**
+   * Change the background color of the apply button when hovering the button
+   * @param{string} color
+   */
+  applyButtonsBgColor = (color) => {
```

Comment:
A method/function should be started with a verb. (ex. `getX()` and `writeToFile()`. So shall we rename this like,
`setApplyButtonBgColor()`